witwidget/notebook/colab/wit.py

```diff
@@ -263,24 +309,96 @@ def _get_element_html(self):
+    i = 0
+    while num_pieces > 0:
+      num_pieces -= 1
+      exs = self.examples[i : i + self.SLICE_SIZE]
+      piece = {'examples': exs, 'countdown': num_pieces}
+      output.eval_js(eval_js_str.format(data=json.dumps(piece)))
+      i += self.SLICE_SIZE
 
   def infer(self):
     try:
       inferences = base.WitWidgetBase.infer_impl(self)
-      output.eval_js("""inferenceCallback({inferences})""".format(
-          inferences=json.dumps(inferences)))
+      indices = inferences['inferences']['indices'][:]
+      inferences['inferences']['indices'] = []
+      res2 = []
+      extra = {}
+      extra2 = {}
+      model_inference = inferences['inferences']['results'][0]
+      if ('extra_outputs' in inferences and len(inferences['extra_outputs']) and
+          inferences['extra_outputs'][0]):
+        for key in inferences['extra_outputs'][0]:
+          extra[key] = inferences['extra_outputs'][0][key][:]
+          inferences['extra_outputs'][0][key] = []
+      if 'classificationResult' in model_inference:
+        res = model_inference['classificationResult']['classifications'][:]
+        model_inference['classificationResult']['classifications'] = []
+      else:
+        res = model_inference['regressionResult']['regressions'][:]
+        model_inference['regressionResult']['regressions'] = []
+      
+      if len(inferences['inferences']['results']) > 1:
+        if ('extra_outputs' in inferences and
+            len(inferences['extra_outputs']) > 1 and
+            inferences['extra_outputs'][1]):
+          for key in inferences['extra_outputs'][1]:
+            extra2[key] = inferences['extra_outputs'][1][key][:]
+            inferences['extra_outputs'][1][key] = []
+        model_2_inference = inferences['inferences']['results'][1]
+        if 'classificationResult' in model_2_inference:
+          res2 = model_2_inference['classificationResult']['classifications'][:]
+          model_2_inference['classificationResult']['classifications'] = []
+        else:
+          res2 = model_2_inference['regressionResult']['regressions'][:]
+          model_2_inference['regressionResult']['regressions'] = []
+
+      i = 0
+      num_pieces = math.ceil(len(indices) / self.SLICE_SIZE)
+
+      # Loop over each piece to send.
+      while num_pieces > 0:
+        num_pieces -= 1
+        piece = [res[i : i + self.SLICE_SIZE]]
+        extra_piece = [{}]
+        for key in extra:
+          extra_piece[0][key] = extra[key][i : i + self.SLICE_SIZE]
+        if res2:
+          piece.append(res2[i : i + self.SLICE_SIZE])
+          extra_piece.append({})
+          for key in extra2:
+            extra_piece[1][key] = extra2[key][i : i + self.SLICE_SIZE]
+        ind_piece = indices[i : i + self.SLICE_SIZE]
+        data = {'results': piece, 'indices': ind_piece, 'extra': extra_piece,
+                'countdown': num_pieces}
+        # For the first segment to send, also send the blank inferences
```

Comment:
It would be better if you made this comment more clear by saying you empty inferences array above. I think you empty all fields in inferences above and send an empty inferences dictionary to have the correct nested structure.

Reply:
done